mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -371,3 +445,46 @@ def set_binning_visibility(self) -> None:
         self.bin_size_spinBox.setHidden(hide_binning)
         self.bin_step_label.setHidden(hide_binning)
         self.bin_step_spinBox.setHidden(hide_binning)
+
+    def set_roi_properties(self) -> None:
+        for spinbox in self.roiPropertiesSpinBoxes.values():
+            spinbox.setEnabled(True)
+        if self.presenter.export_mode == ExportMode.IMAGE_MODE:
+            self.current_roi = ROI_RITS
+        current_roi = self.presenter.model.get_roi(self.current_roi)
+        self.roiPropertiesGroupBox.setTitle(f"Roi Properties: {self.current_roi}")
+        roi_iter_order = ["Left", "Top", "Right", "Bottom"]
+        for row, pos in enumerate(current_roi):
+            with QSignalBlocker(self.roiPropertiesSpinBoxes[roi_iter_order[row]]):
+                self.roiPropertiesSpinBoxes[roi_iter_order[row]].setValue(pos)
+        self.set_roi_spinbox_ranges()
+        self.presenter.redraw_spectrum(self.current_roi)
+        self.roiPropertiesLabels["Width"].setText(str(current_roi.width))
+        self.roiPropertiesLabels["Height"].setText(str(current_roi.height))
+
+    def adjust_roi(self) -> None:
+        roi_iter_order = ["Left", "Top", "Right", "Bottom"]
+        new_points = [self.roiPropertiesSpinBoxes[prop].value() for prop in roi_iter_order]
+        new_roi = SensibleROI().from_list(new_points)
+        self.presenter.model.set_roi(self.current_roi, new_roi)
+        self.spectrum_widget.adjust_roi(new_roi, self.current_roi)
+
+    def set_roi_spinbox_ranges(self):
+        self.roiPropertiesSpinBoxes["Left"].setMaximum(self.roiPropertiesSpinBoxes["Right"].value() - 1)
+        self.roiPropertiesSpinBoxes["Right"].setMinimum(self.roiPropertiesSpinBoxes["Left"].value() + 1)
+        self.roiPropertiesSpinBoxes["Top"].setMaximum(self.roiPropertiesSpinBoxes["Bottom"].value() - 1)
+        self.roiPropertiesSpinBoxes["Bottom"].setMinimum(self.roiPropertiesSpinBoxes["Top"].value() + 1)
+
+    def disable_roi_properties(self):
+        self.roiPropertiesGroupBox.setTitle("Roi Properties: None selected")
+        self.last_clicked_roi = "roi"
```

Comment:
When changing samples within the Image mode, the new ROI generated falls out of sync with the ROI that the spin boxes are watching such that when I resize the ROI within the image view, the spin box values do not update.